File: serviciu_bilete/src/main/java/com/pos/security/AuthenticatedUser.java
package com.pos.security;

public record AuthenticatedUser(int userId, String role) {

    public boolean isAdmin() {
        return "admin".equals(role);
    }

    public boolean isOwnerEvent() {
        return "owner-event".equals(role);
    }

    public boolean isClient() {
        return "client".equals(role);
    }

    public boolean canManageResource(Integer resourceOwnerId) {
        if (isAdmin()) {
            return true;
        }
        if (isOwnerEvent() && resourceOwnerId != null) {
            return resourceOwnerId == userId;
        }
        return false;
    }
}

File: serviciu_clienti/src/main/kotlin/com/pos/serviciu_clienti/security/AuthenticatedUser.kt
package com.pos.serviciu_clienti.security

data class AuthenticatedUser(
    val userId: Int,
    val role: String
) {
    fun isAdmin(): Boolean = role == "admin"

    fun isOwnerEvent(): Boolean = role == "owner-event"

    fun isClient(): Boolean = role == "client"

    fun canManageClient(clientIdmUserId: Int?): Boolean {
        if (isAdmin()) return true
        if (isClient() && clientIdmUserId != null) {
            return clientIdmUserId == userId
        }
        return false
    }
}
